src/SourceLocator/Ast/PhpParserLocator.php

```diff
@@ -0,0 +1,93 @@
+<?php
+declare(strict_types=1);
+
+namespace Roave\BetterReflection\SourceLocator\Ast;
+
+use Roave\BetterReflection\Identifier\Identifier;
+use Roave\BetterReflection\Identifier\IdentifierType;
+use Roave\BetterReflection\SourceLocator\Ast\Parser\MemoizingParser;
+use Roave\BetterReflection\SourceLocator\Ast\Strategy\NodeToReflection;
+use Roave\BetterReflection\Reflection\Reflection;
+use Roave\BetterReflection\Reflector\Reflector;
+use Roave\BetterReflection\Reflector\Exception\IdentifierNotFound;
+use Roave\BetterReflection\SourceLocator\Located\LocatedSource;
+use PhpParser\Parser;
+use PhpParser\ParserFactory;
+
+/**
+ * @internal
+ */
+final class PhpParserLocator implements Locator
```

Comment:
This is just split out from `Locator` - I don't know if this change should be reverted.